### PR TITLE
Switch some crates to the 2018 edition 

### PR DESCRIPTION
**Pull request by SimonSapin**
This is the subset of https://github.com/servo/servo/pull/22083 that doesn’t affect Gecko at all, so it isn’t blocked.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22086)

### CODE REVIEW

File: components/canvas_traits/webgl.rs [hunk 1]
```diff
@@ -342,7 +342,7 @@ macro_rules! define_resource_id {
             fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
                 where D: ::serde::Deserializer<'de>
             {
-                let id = try!(u32::deserialize(deserializer));
+                let id = r#try!(u32::deserialize(deserializer));
```
Comment (KiChjang) [inline components/canvas_traits/webgl.rs]: Can this just be
```suggestion
                let id = u32::deserialize(deserializer)?;
```

Comment (SimonSapin) [reply components/canvas_traits/webgl.rs]: Thanks!

**Comment by jdm** [comment]
The windows build is unhappy.

**Comment by SimonSapin** [comment]
Fixed up macOS and Windows.

@bors-servo try

**Comment by SimonSapin** [comment]
I believe this Android x86 emulator test failure is intermittent, clicking "Actions" then "Retrigger Task" produced a green task: https://tools.taskcluster.net/groups/VUgXtYAwTPKWbpjaDZ72BA/tasks/I2Lh3hmtQq-7alaHFQPP9A/details

**Comment by SimonSapin** [comment]
This is almost entirely mechanical, let’s rubber stamp.

@bors-servo r+

**Comment by SimonSapin** [comment]
@bors-servo retry

*  #19995

**Pull request merged**